include/geos/geom/FixedSizeCoordinateSequence.h

```diff
@@ -0,0 +1,129 @@
+/**********************************************************************
+ *
+ * GEOS - Geometry Engine Open Source
+ * http://geos.osgeo.org
+ *
+ * Copyright (C) 2019 Daniel Baston
+ *
+ * This is free software; you can redistribute and/or modify it under
+ * the terms of the GNU Lesser General Public Licence as published
+ * by the Free Software Foundation.
+ * See the COPYING file for more information.
+ *
+ **********************************************************************/
+
+#ifndef GEOS_GEOM_FIXEDSIZECOORDINATESEQUENCE_H
+#define GEOS_GEOM_FIXEDSIZECOORDINATESEQUENCE_H
+
+#include <geos/geom/Coordinate.h>
+#include <geos/geom/CoordinateFilter.h>
+#include <geos/geom/CoordinateSequence.h>
+#include <geos/util.h>
+
+#include <algorithm>
+#include <array>
+#include <memory>
+#include <sstream>
+#include <vector>
+
+namespace geos {
+namespace geom {
+
+    template<size_t N>
+    class FixedSizeCoordinateSequence : public CoordinateSequence {
+    public:
+        explicit FixedSizeCoordinateSequence(size_t dimension_in = 0) : dimension(dimension_in) {}
+
+        std::unique_ptr<CoordinateSequence> clone() const override {
```

Comment:
IMHO, this should be declared `clone() const final`, unless there really is need to allow deriving new types from `FixedSizeCoordinateSequence` class.

Reply:
Done, thanks. I guess the benefit is to potentially allow compiler devirtualization?